From 0b0d37a1c7cd11cbc25a91787b9f08c407559146 Mon Sep 17 00:00:00 2001
From: Matthew Schwartz <matthew.schwartz@linux.dev>
Date: Fri, 27 Mar 2026 11:49:02 -0700
Subject: [PATCH 096/188] hid-asus-ally: Separate trigger rumble from main
 rumble

The driver copies main rumble values into additional rumble fields, which
are now used for trigger rumble on the ASUS ROG Xbox Ally X. This causes
both the main rumble and the new trigger rumble motors to fire at the
same intensity on ASUS ROG Xbox Ally X, which effectively double rumble
intensity on that model.

This is already a no-op on older Ally models that do not have trigger
rumble and we don't use the trigger rumble for anything on SteamOS, so
we can just disable it by separating out the trigger rumble fields.

Signed-off-by: Matthew Schwartz <matthew.schwartz@linux.dev>
---
 drivers/hid/hid-asus-ally.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/drivers/hid/hid-asus-ally.c b/drivers/hid/hid-asus-ally.c
index 8b51560fd..dc7d5de94 100644
--- a/drivers/hid/hid-asus-ally.c
+++ b/drivers/hid/hid-asus-ally.c
@@ -1541,11 +1541,8 @@ static void ally_x_work(struct work_struct *work)
 	update_qam = ally_x->update_qam_btn;
 	spin_unlock_irqrestore(&ally_x->lock, flags);
 
-	if (update_ff && ff_report) {
-		ff_report->ff.magnitude_left = ff_report->ff.magnitude_strong;
-		ff_report->ff.magnitude_right = ff_report->ff.magnitude_weak;
+	if (update_ff && ff_report)
 		asus_dev_set_report(ally_x->hdev, (u8 *)ff_report, sizeof(*ff_report));
-	}
 	kfree(ff_report);
 
 	if (update_qam) {
-- 
2.55.0

